5.2.cup_of_join.py

```diff
@@ -0,0 +1,16 @@
+# Author: Omer Gertler
+
+from functools import reduce
+
+
+def join(*lists, sep='-'):
+    if len(lists) > 0:
+        res = reduce(lambda x, y: x + ([sep] + y), lists, [])
```

Comment:
Consider changing 'res' to 'result', 'sep' to 'separate', and maybe even x, y to first_list, second_list, imagine seeing the code as part of 500 lines of code, it could be unclear.